src/placeos-resource.cr

```diff
@@ -2,16 +2,32 @@ require "deque"
 require "log_helper"
 require "promise"
 require "rethinkdb-orm"
-require "simple_retry"
+require "retriable"
 
 # Internally abstracts data event streams.
 #
 abstract class PlaceOS::Resource(T)
   Log = ::Log.for(self)
 
-  alias Error = NamedTuple(name: String, reason: String)
   alias Action = RethinkORM::Changefeed::Event
 
+  struct Error
+    getter name : String, reason : String
+
+    def initialize(name, reason)
+      @name = name || ""
+      @reason = reason || ""
+    end
```

Comment:
```suggestion
    def initialize(name="", reason="")
    end
```

Reply:
i want the arguments to be passed, hence defaults should not be used.
the pattern here allows nils to be passed and set to `""` if  `nil` was passed.